rpc/handlers.go

```diff
@@ -450,3 +451,40 @@ func (h *Handler) Syncing() (*SyncState, *jsonrpc.Error) {
 		},
 	}, nil
 }
+
+// Nonce returns the nonce associated with the given address in the given block number
+//
+// It follows the specification defined here:
+// https://github.com/starkware-libs/starknet-specs/blob/a789ccc3432c57777beceaa53a34a7ae2f25fda0/api/starknet_api_openrpc.json#L633
+func (h *Handler) Nonce(id *BlockID, address *felt.Felt) (*felt.Felt, *jsonrpc.Error) {
+	var stateReader core.StateReader
+	var stateCloser func() error
+	var err error
+
+	defer func() {
+		if stateCloser != nil {
+			stateCloser()
+		}
+	}()
+
+	if id.Latest {
+		stateReader, stateCloser, err = h.bcReader.HeadState()
+	} else if id.Pending {
+		err = ErrPendingNotSupported
+	} else if id.Hash != nil {
+		stateReader, stateCloser, err = h.bcReader.StateAtBlockHash(id.Hash)
+	} else {
+		stateReader, stateCloser, err = h.bcReader.StateAtBlockNumber(id.Number)
+	}
+
+	if err != nil {
+		return nil, ErrBlockNotFound
+	}
```

Comment:
Let's make this a function, it will be reused for sure.

Reply:
like; https://github.com/NethermindEth/juno/pull/723/files#diff-1f724c5e433be46bfd8dc0962d1796f0dce2a5e22253fe82f69ab6a1eb8df278R456-R467